waspc/packages/wasp-config/src/appSpec.ts

```diff
@@ -0,0 +1,210 @@
+/** This module is a mirror implementation of AppSpec Decls in TypeScript.
+ * The original implemention is in Haskell (waspc).
+ *
+ * IMPORTANT: Do not change this file without updating the AppSpec in waspc.
```

Comment:
I'll add this comment there too.

It would be nice if AppSpec came from a single specification independent of TS and HS.

Reply:
We could generate it from Haskell! Shouldn't do it now, but it should be doable in the future. That said, I am not sure if that is smart or not. We should have our hands free here in TS to do as much type level stuff as we want and to also adjust things as we like. So ok, not all should be genreated. But maybe a small part of it should be generated, enough to ensure that two things are in sync. Which maybe is this very file at the end, yeah all right.